### PR TITLE
Closes #3631: refactor-utilMsg-to-remove-registerND-annotation

### LINKED ISSUE #3631: refactor UtilMsg to remove registerND annotation
Refactor UtilMsg to remove registerND annotation.

### PR DESCRIPTION
**Pull request by ajpotts**
Refactored `UtilMsg` to remove `@arkouda.registerND` annotation.  Also added some extra unit tests to verify the unsupported types throw `RuntimeError`s.

Closes #3631: refactor-utilMsg-to-remove-registerND-annotation

### CODE REVIEW

File: src/UtilMsg.chpl [hunk 1]
```diff
@@ -21,127 +23,81 @@ module UtilMsg {
...
+          forall axisSliceIdx in domOffAxis(d, axis) {
+            const slice = domOnAxis(diffSubDom, tuplify(axisSliceIdx), axis);
+
+            for i in slice {
+              var idxp = tuplify(i);
+              idxp[axis] += 1;
+              d1[i] = d2[idxp] - d2[i];
             }
           }
-        } // d2 deinit here
```
Comment (jeremiah-corrado) [inline src/UtilMsg.chpl:122]: could you keep this comment (it explains why d2's declaration needs to be in a nested scope).

File: src/UtilMsg.chpl [hunk 2]
```diff
@@ -165,71 +121,52 @@ module UtilMsg {
...
-            beforeSlice[i] = 0..<outShape[i];
-            afterSlice[i] = 0..<outShape[i];
-          }
+  @arkouda.instantiateAndRegister
+  proc pad(cmd: string, msgArgs: borrowed MessageArgs, st: borrowed SymTab, type array_dtype, param array_nd: int): MsgTuple throws 
+    where (array_dtype == int) || (array_dtype == uint(8)) || (array_dtype == uint(64)) || (array_dtype == real) || (array_dtype == bool) {
+
+    const e = st[msgArgs["name"]]: SymEntry(array_dtype, array_nd),
+          padWidthBefore = msgArgs["padWidthBefore"].getTuple(array_nd),
+          padWidthAfter = msgArgs["padWidthAfter"].getTuple(array_nd),
```
Comment (jeremiah-corrado) [inline src/UtilMsg.chpl:130]: Can you use `toScalarTuple(int, array_nd)` here? The plan is to get rid of `getTuple` eventually.

**Review by jeremiah-corrado — APPROVED** [review]
Looks good! Just a couple of suggestions:

File: src/UtilMsg.chpl (hunk 1 above)
Comment (ajpotts) [reply src/UtilMsg.chpl:122]: I put it back!

File: src/UtilMsg.chpl (hunk 2 above)
Comment (ajpotts) [reply src/UtilMsg.chpl]: I fixed this.  Thanks for pointing it out.

**Review by stress-tess — APPROVED** [review]

**Pull request merged**